mobile-save-for-later/src/main/scala/com/gu/sfl/util/Compression.scala

```diff
@@ -0,0 +1,59 @@
+package com.gu.sfl.util
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, InputStream, OutputStream}
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Base64
+import java.util.zip.Deflater.BEST_COMPRESSION
+import java.util.zip.{Deflater, GZIPInputStream, GZIPOutputStream}
+
+import com.gu.sfl.util.Base64Helper.{base64Decode, base64Encode}
+import org.apache.commons.io.IOUtils
+
+object Base64Helper {
+  private val encoder: Base64.Encoder = Base64.getEncoder
+  private val decoder: Base64.Decoder = Base64.getDecoder
+
+  def base64Encode(outputStream: OutputStream): OutputStream = encoder.wrap(outputStream)
+
+  def base64Decode(inputStream: InputStream): InputStream = decoder.wrap(inputStream)
+}
+class LevelModifiableGZIPOutputStream(outputStream: OutputStream) extends GZIPOutputStream(outputStream) {
+  def setLevel(level: Int): Unit = {
+    this.`def` = new Deflater(level, true)
```

Comment:
Is this because 'def' is a keyword in scala, but not java?

Reply:
Yes